### PR TITLE
Create deny endpoint

### PR DESCRIPTION
**Pull request by Wirt4**
This PR fills in the api endpoint for denying a friend request. `/friends/deny` does two things:
- removes the friend request from the database
- triggers a call on the friends channel with the event "deny_friend"
FirendRequestSidebarOptions has been updated to listen for "deny_friend" and decrease the request count when that occurs.

previously it was just stubbed out, which caused not found errors to be thrown when the user hit it on the page.

I also removed some tests that evaluated implementation instead of behavior.

In summary: 
 - added a new api route: /friends/deny
 - `/friends/deny` edits the DB according and triggers a pusher message
 - added a listener for the pusher message in FriendRequestSidebarOptions

### CODE REVIEW

File: __tests__/app/api/friends/deny/route.test.ts [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+jest.mock("@/lib/dbWrapper", () => ({
+    removeDbEntry: jest.fn(),
+}));
+
+jest.mock("@/lib/pusher",()=>({
+    getPusherServer: jest.fn()
+}));
+
+
```
Comment (Wirt4) [inline __tests__/app/api/friends/deny/route.test.ts:18]: remove one newline

File: __tests__/components/FriendRequestSidebarOptions/helpers.test.ts [hunk 2]
```diff
@@ -190,55 +120,52 @@ describe('PusherClientHandler return tests', () => {
         client = new PusherClientHandler('stub_id', 0)
     })
 
-    test('return is a function, it should call unbind for friend requests',()=>{
-        const func = client.subscribeToPusher(jest.fn())
-        func()
-        expect(requestsUnbindSpy).toHaveBeenCalledWith('incoming_friend_requests', expect.anything())
-    })
+    //TODO: test here for "deny_friend"
```
Comment (Wirt4) [inline __tests__/components/FriendRequestSidebarOptions/helpers.test.ts:123]: remove comment

File: src/app/api/friends/deny/route.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import myGetServerSession from "@/lib/myGetServerSession";
+import {z} from "zod";
+import {removeDbEntry} from "@/lib/dbWrapper";
+import QueryBuilder from "@/lib/queryBuilder";
+import {getPusherServer} from "@/lib/pusher";
+
+export async function POST(req: Request) {
```
Comment (Wirt4) [inline src/app/api/friends/deny/route.ts:7]: This is much cleaner than the object abstraction. I like it.

File: src/components/friendRequestSidebarOptions/helpers.ts [hunk 4]
```diff
@@ -19,6 +19,7 @@ export default class PusherClientHandler{
             requests:  QueryBuilder.incomingFriendRequestsPusher(this.id),
             friends: QueryBuilder.friendsPusher(this.id)
         }
+
```
Comment (Wirt4) [inline src/components/friendRequestSidebarOptions/helpers.ts:22]: ditch this

File: __tests__/components/FriendRequestSidebarOptions/helpers.test.ts [hunk 5]
```diff
@@ -190,55 +120,52 @@ describe('PusherClientHandler return tests', () => {
...
 
-    test('return is a function, it should call unbind for new_friend',()=>{
+    test('Given that the return is a function: When that function is called, ' +
+        'then the new_friend event is unbound with the decrementCount callback',()=>{
```
Comment (Wirt4) [inline __tests__/components/FriendRequestSidebarOptions/helpers.test.ts:126]: let's bite the bullet and make the entire statement for each of these descriptions on one line. If it's wordy, it may be a signal to break up the evaluation into more granular tests.
